### PR TITLE
[13.0][FIX] account: consider mail.alias relation to invoices

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I'm afraid this doesn't serve, as the alias also contains some applied default values that will make this fail anyway. We need to call the recreation of the mail.alias by ORM for each journal on post, and remove this bulk renaming.

**Comment by MiquelRForgeFlow** [comment]
How to call the recreation of the mail.alias?

**Comment by pedrobaeza** [comment]
Don't know yet. Check how is done in the code.

**Comment by kos94ok-3D** [comment]
@MiquelRForgeFlow Maybe it will help you. account journal have `_update_mail_alias` function

**Comment by MiquelRForgeFlow** [comment]
Thanks @kos94ok-3D , the `_update_mail_alias` method helped me understand what needed to be updated. Also I tested it in a migrated db of a client :)

**Comment by pedrobaeza** [comment]
This is still not changing the default values in alias.

**Comment by pedrobaeza** [comment]
Just call `_update_mail_alias` in all journals with alias_name.

**Comment by MiquelRForgeFlow** [comment]
In `_update_mail_alias` method, the alias gets updated with:

```
    'alias_defaults': {'type': type == 'purchase' and 'in_invoice' or 'out_invoice', 'company_id': self.company_id.id, 'journal_id': self.id},
    'alias_parent_thread_id': self.id,
    'alias_name': alias_name,
```
As far I can see, none of these values should change or be changed.

**Comment by pedrobaeza** [comment]
Needed code:

```python
journals = env["account.journal"].search([("alias_name", "!=", False)])
for journal in journals:
    journal.alias_id.unlink()
    journal._update_mail_alias({"alias_name": journal.alias_name})
```

**Comment by pedrobaeza** [comment]
In previous alias, `company_id` wasn't present.

**Comment by pedrobaeza** [comment]
And other customization may add more values.

**Comment by pedrobaeza** [comment]
I'm thinking this should be run on `end-migration` due to my last sentence.

**Comment by MiquelRForgeFlow** [comment]
> In previous alias, company_id wasn't present.

False, I checked in v12, and code is the same. That's why my migrated multicompany db has alias names with the company.

> And other customization may add more values.

Ok, that's your unique valid point. Fuck customizations.

File: addons/account/migrations/13.0.1.1/end-migration.py [hunk 1]
```diff
@@ -44,6 +44,15 @@ def add_default_account_data(env):
             company[company_field] = account.id
 
 
+def update_mail_alias_for_moves(env):
+    # Done here just in case alias customizations exist
+    journals = env["account.journal"].search([("alias_id", "!=", False)])
```
Comment (MiquelRForgeFlow) [inline addons/account/migrations/13.0.1.1/end-migration.py]:
```suggestion
    journals = env["account.journal"].with_context(active_test=False).search([("alias_id", "!=", False)])
```
?

**Comment by pedrobaeza** [comment]
About the company, sorry, I meant `journal_id` value.

File: addons/account/migrations/13.0.1.1/end-migration.py (hunk 1 above)
Comment (pedrobaeza) [reply addons/account/migrations/13.0.1.1/end-migration.py]: Yes, I have just checked that disable journals still preserve the alias.

File: addons/account/migrations/13.0.1.1/end-migration.py [hunk 2]
```diff
@@ -44,6 +44,16 @@ def add_default_account_data(env):
             company[company_field] = account.id
 
 
+def update_mail_alias_for_moves(env):
+    # Done here just in case alias customizations exist
+    journals = env["account.journal"].with_context(
+        active_test=False).search([("alias_id", "!=", False)])
```
Comment (pedrobaeza) [inline addons/account/migrations/13.0.1.1/end-migration.py:50]: As it's the important data

```suggestion
        active_test=False).search([("alias_name", "!=", False)])
```

Comment (pedrobaeza) [reply addons/account/migrations/13.0.1.1/end-migration.py:50]: Ah, no, forget it

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**